### PR TITLE
Uses Array.init(unsafeUninitializedCapacity:,initializingWith:)

### PR DESCRIPTION
**Pull request by valeriyvan**
Checklist:
- [ ] Correct file headers (see CONTRIBUTING.md).
- [ ] Formatted with [SwiftFormat](https://github.com/nicklockwood/SwiftFormat).
- [ ] Tests added.

Changes proposed in this pull request:
- Uses Array.init(unsafeUninitializedCapacity:,initializingWith:) where possible in Rabbit.swift

### CODE REVIEW

**Comment by krzyzanowskim** [comment]
1. Does that improve performance?
1. Code may expect zeroes (like from calloc()), did you look at that?

This is Swift 5.1. I have to branch off Swift 5.0 first, then start with adding Swift 5.1 changes.

**Comment by valeriyvan** [comment]
> 1. Does that improve performance?
Yes. It avoids initialising of arrays where this initialisation is immediately overwritten. 

> Code may expect zeroes (like from calloc()), did you look at that?
I left such places untouched.

> This is Swift 5.1. I have to branch off Swift 5.0 first, then start with adding Swift 5.1 changes.
That's fine.

**Comment by krzyzanowskim** [comment]
Can you post bench numbers? Those are critical performance parts that I would rather avoid to regress.

**Comment by valeriyvan** [comment]
Refactor shows ~5% performance gain of `RabbitTestsPerf` test (1.462 sec vs 1.534 sec).

**Comment by krzyzanowskim** [comment]
I tested it with `testRabbitPerformance` where 
```
let message = Array<UInt8>(repeating: 7, count: (8 * 1024 * 1024) * 1)
```

### valeriyvan-unsafeUninitializedCapacity
Time: 0.476 sec
Time: 0.488 sec
Time: 0.473 sec
Time: 0.504 sec
Time: 0.494 sec
Time: 0.464 sec

### master
Time: 0.522 sec
Time: 0.492 sec
Time: 0.496 sec
Time: 0.470 sec
Time: 0.471 sec
Time: 0.484 sec
Time: 0.476 sec

The results look the same to me.
I didn't notice a stable performance decrease nor improvement for that sample.

File: Sources/CryptoSwift/Rabbit.swift [hunk 1]
```diff
@@ -147,16 +149,17 @@ public final class Rabbit: BlockCipher {
...
+    self.x = Array<UInt32>(unsafeUninitializedCapacity: 8) { newX, count in
+      newX[0] = self.g(0) &+ rotateLeft(self.g(7), by: 16) &+ rotateLeft(self.g(6), by: 16)
+      newX[1] = self.g(1) &+ rotateLeft(self.g(0), by: 8) &+ self.g(7)
+      newX[2] = self.g(2) &+ rotateLeft(self.g(1), by: 16) &+ rotateLeft(self.g(0), by: 16)
+      newX[3] = self.g(3) &+ rotateLeft(self.g(2), by: 8) &+ self.g(1)
+      newX[4] = self.g(4) &+ rotateLeft(self.g(3), by: 16) &+ rotateLeft(self.g(2), by: 16)
+      newX[5] = self.g(5) &+ rotateLeft(self.g(4), by: 8) &+ self.g(3)
+      newX[6] = self.g(6) &+ rotateLeft(self.g(5), by: 16) &+ rotateLeft(self.g(4), by: 16)
+      newX[7] = self.g(7) &+ rotateLeft(self.g(6), by: 8) &+ self.g(5)
+      count = 8
```
Comment (krzyzanowskim) [inline Sources/CryptoSwift/Rabbit.swift]: why do we need this line?

**Comment by krzyzanowskim** [comment]
Let's give it a try. Thanks you @valeriyvan 
Sorry for the delay in all of it

**Pull request merged**